'default' is not defined in typeof import

**Repro**

<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.

Please consider creating an isolated reproduction repo to make it easy for the volunteer maintainers debug your issue.
-->

```JavaScript
module.exports = {
  root: true,
  extends: '@react-native-community',
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
};

```

```TS
export type RootAction = ActionType<typeof import('./src/store/actions').default>;
export type AppState = ReturnType<typeof import('./src/store/reducers').default>;
```

<!--
Also include your tsconfig, if you're using type-aware linting
-->

**Expected Result**

No errors here like it was before

**Actual Result**

```
/.../types.d.ts
  4:74  error  'default' is not defined  no-undef
  5:73  error  'default' is not defined  no-undef
```

**Additional Info**

In this files (actions and reducers) I have default export

```
const appReducer = combineReducers({
// ...
});

export default appReducer;
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.14.2` |
| `TypeScript`                | `4.1.3` |
| `ESLint`                    | `7.19.0` |
| `node`                      | `12.18.4` |

--- a/packages/eslint-plugin/tests/eslint-rules/no-undef.test.ts
+++ b/packages/eslint-plugin/tests/eslint-rules/no-undef.test.ts
@@ -235,6 +235,10 @@ declare function deco(...param: any): (...param: any) => any;
 })
 class Foo {}
     `,
+    // https://github.com/typescript-eslint/typescript-eslint/issues/3006
+    `
+export type AppState = typeof import('./src/store/reducers').default;
+    `,
   ],
   invalid: [
     {

--- a/packages/scope-manager/src/referencer/TypeVisitor.ts
+++ b/packages/scope-manager/src/referencer/TypeVisitor.ts
@@ -116,6 +116,12 @@ class TypeVisitor extends Visitor {
     this.visitFunctionType(node);
   }
 
+  protected TSImportType(node: TSESTree.TSImportType): void {
+    // the TS parser allows any type to be the parameter, but it's a syntax error - so we can ignore it
+    this.visit(node.typeParameters);
+    // the qualifier is just part of a standard EntityName, so it should not be visited
+  }
+
   protected TSIndexSignature(node: TSESTree.TSIndexSignature): void {
     for (const param of node.parameters) {
       if (param.type === AST_NODE_TYPES.Identifier) {

--- a/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-qualifier.ts
+++ b/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-qualifier.ts
@@ -1,0 +1,1 @@
+type T = import('').default;

--- a/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-qualifier.ts.shot
+++ b/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-qualifier.ts.shot
@@ -1,0 +1,38 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`type-declaration import-type-with-qualifier 1`] = `
+ScopeManager {
+  variables: Array [
+    ImplicitGlobalConstTypeVariable,
+    Variable$2 {
+      defs: Array [
+        TypeDefinition$1 {
+          name: Identifier<"T">,
+          node: TSTypeAliasDeclaration$1,
+        },
+      ],
+      name: "T",
+      references: Array [],
+      isValueVariable: false,
+      isTypeVariable: true,
+    },
+  ],
+  scopes: Array [
+    GlobalScope$1 {
+      block: Program$2,
+      isStrict: false,
+      references: Array [],
+      set: Map {
+        "const" => ImplicitGlobalConstTypeVariable,
+        "T" => Variable$2,
+      },
+      type: "global",
+      upper: null,
+      variables: Array [
+        ImplicitGlobalConstTypeVariable,
+        Variable$2,
+      ],
+    },
+  ],
+}
+`;

--- a/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-type-params.ts
+++ b/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-type-params.ts
@@ -1,0 +1,2 @@
+type Param = string;
+type T = import('').foo<Param>;

--- a/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-type-params.ts.shot
+++ b/packages/scope-manager/tests/fixtures/type-declaration/import-type-with-type-params.ts.shot
@@ -1,0 +1,63 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`type-declaration import-type-with-type-params 1`] = `
+ScopeManager {
+  variables: Array [
+    ImplicitGlobalConstTypeVariable,
+    Variable$2 {
+      defs: Array [
+        TypeDefinition$1 {
+          name: Identifier<"Param">,
+          node: TSTypeAliasDeclaration$1,
+        },
+      ],
+      name: "Param",
+      references: Array [
+        Reference$1 {
+          identifier: Identifier<"Param">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: false,
+          isWrite: false,
+          resolved: Variable$2,
+        },
+      ],
+      isValueVariable: false,
+      isTypeVariable: true,
+    },
+    Variable$3 {
+      defs: Array [
+        TypeDefinition$2 {
+          name: Identifier<"T">,
+          node: TSTypeAliasDeclaration$2,
+        },
+      ],
+      name: "T",
+      references: Array [],
+      isValueVariable: false,
+      isTypeVariable: true,
+    },
+  ],
+  scopes: Array [
+    GlobalScope$1 {
+      block: Program$3,
+      isStrict: false,
+      references: Array [
+        Reference$1,
+      ],
+      set: Map {
+        "const" => ImplicitGlobalConstTypeVariable,
+        "Param" => Variable$2,
+        "T" => Variable$3,
+      },
+      type: "global",
+      upper: null,
+      variables: Array [
+        ImplicitGlobalConstTypeVariable,
+        Variable$2,
+        Variable$3,
+      ],
+    },
+  ],
+}
+`;

--- a/packages/scope-manager/tests/fixtures/type-declaration/import-type.ts
+++ b/packages/scope-manager/tests/fixtures/type-declaration/import-type.ts
@@ -1,0 +1,1 @@
+type T = import('');

--- a/packages/scope-manager/tests/fixtures/type-declaration/import-type.ts.shot
+++ b/packages/scope-manager/tests/fixtures/type-declaration/import-type.ts.shot
@@ -1,0 +1,38 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`type-declaration import-type 1`] = `
+ScopeManager {
+  variables: Array [
+    ImplicitGlobalConstTypeVariable,
+    Variable$2 {
+      defs: Array [
+        TypeDefinition$1 {
+          name: Identifier<"T">,
+          node: TSTypeAliasDeclaration$1,
+        },
+      ],
+      name: "T",
+      references: Array [],
+      isValueVariable: false,
+      isTypeVariable: true,
+    },
+  ],
+  scopes: Array [
+    GlobalScope$1 {
+      block: Program$2,
+      isStrict: false,
+      references: Array [],
+      set: Map {
+        "const" => ImplicitGlobalConstTypeVariable,
+        "T" => Variable$2,
+      },
+      type: "global",
+      upper: null,
+      variables: Array [
+        ImplicitGlobalConstTypeVariable,
+        Variable$2,
+      ],
+    },
+  ],
+}
+`;

